src/components/modules/paste.ts

```diff
@@ -145,8 +145,27 @@ export default class Paste extends Module {
    * @public
    */
   public async prepare(): Promise<void> {
-    this.setCallback();
-    this.processTools();
+    this.toggleReadOnly(this.config.readOnly);
+  }
+
+  /**
+   * Set read-only state
+   *
+   * @param {boolean} readOnlyEnabled - read only flag value
+   */
+  public toggleReadOnly(readOnlyEnabled: boolean): void {
+    if (!readOnlyEnabled) {
+      this.processTools();
+      this.setCallback();
+    } else {
+      this.unsetCallback();
+
+      this.toolsTags = {};
+      this.tagsByTool = {};
+      this.toolsPatterns = [];
+      this.toolsFiles = {};
+      this.exceptionList = [];
```

Comment:
Do we really need to empty all of these properties?

Maybe it's better to add cache check in the `processTool` method

Reply:
fixed